### PR TITLE
Init synpress and playwright

### PR DESCRIPTION
**Pull request by Spacesai1or**

### CODE REVIEW

**Comment by Spacesai1or** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7148** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7148" target="_blank">Graphite</a> 
    * **PR #7149** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7149" target="_blank">Graphite</a> 
      * **PR #7150** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7150" target="_blank">Graphite</a>  👈
        * **PR #7151** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7151" target="_blank">Graphite</a> 
          * **PR #7152** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7152" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7150?utm_source=stack-comment).

File: ufm-test-services/metamask/package.json [hunk 1]
```diff
@@ -12,9 +12,15 @@
   "type": "module",
   "scripts": {
     "lint": "prettier --check .",
-    "lint:fix": "prettier --write ."
+    "lint:fix": "prettier --write .",
+    "start:metamask-dapp": "npx static-server node_modules/@metamask/test-dapp/dist --port 9011"
```
Comment (roninjin10) [inline ufm-test-services/metamask/package.json:17]: very nice!

File: ufm-test-services/metamask/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    /* Test against branded browsers. */
+    // {
+    //   name: 'Microsoft Edge',
+    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
+    // },
+    // {
+    //   name: 'Google Chrome',
+    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
+    // },
```
Comment (roninjin10) [inline ufm-test-services/metamask/playwright.config.ts]:
```suggestion
```
I think it's fine to only run against chromium in this case

File: ufm-test-services/metamask/playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  ],
+
+  /* Run your local dev server before starting the tests */
+  webServer: {
+    command: 'pnpm run start:metamask-dapp',
+    url: process.env.METAMASK_DAPP_URL,
+    reuseExistingServer: false,
```
Comment (roninjin10) [inline ufm-test-services/metamask/playwright.config.ts:45]:
```suggestion
    reuseExistingServer: true,
```
I think it's usually useful to have this on so you can manually test it but up to you what you prefer

File: ufm-test-services/metamask/tests/testWithSynpress.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+			secretWordsOrPrivateKey: process.env.METAMASK_SECRET_WORDS_OR_PRIVATEKEY,
+			network: process.env.METAMASK_NETWORK,
+			password: process.env.METAMASK_PASSWORD,
+			enableAdvancedSettings: true,
+		})
+		await use(context)
+		await context.close()
+	},
+})
+export { expect }
```
Comment (roninjin10) [inline ufm-test-services/metamask/tests/testWithSynpress.ts]:
```suggestion
export { expect }

```
nit: eol

File: ufm-test-services/metamask/tests/metamask.spec.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import { testWithSynpress } from './testWithSynpress'
+import { test } from '@playwright/test'
+
+testWithSynpress('should be able to read', async ({ page }) => {
+	await page.goto('http://localhost:9011')
+})
```
Comment (roninjin10) [inline ufm-test-services/metamask/tests/metamask.spec.ts]:
```suggestion
})

```
nit: eol

**Review by roninjin10 — APPROVED** [review]
lgtm

File: ufm-test-services/metamask/playwright.config.ts (hunk 2 above)
Comment (Spacesai1or) [reply ufm-test-services/metamask/playwright.config.ts]: Removed them

File: ufm-test-services/metamask/playwright.config.ts (hunk 3 above)
Comment (Spacesai1or) [reply ufm-test-services/metamask/playwright.config.ts:45]: I have this off mostly for sanity reasons (clean slate every time you run), and in actual use of the test service, everything is ephemeral inside the docker container anyways

**Pull request merged**